elkodon_bb/container/src/byte_string.rs

```diff
@@ -33,13 +33,14 @@ use elkodon_bb_log::{fail, fatal_panic};
 ///  * The string must be '\0' (null) terminated.
 ///
 pub unsafe fn strlen(ptr: *mut std::ffi::c_char) -> usize {
+    const NULL_TERMINATION: std::ffi::c_char = 0;
     for i in 0..isize::MAX {
-        if *ptr.offset(i) as u8 == b'\0' {
+        if *ptr.offset(i) == NULL_TERMINATION {
             return i as usize;
         }
```

Comment:
It seems `c_char` on Linux aarch64 is already a `u8` which creates quite a lot clippy warnings regarding unnecessary casts. I started fixing a few but noticed that it is more than just a few.

Reply:
Oh I think the rust idiomatic way would be to use `is_null` (https://doc.rust-lang.org/std/primitive.pointer.html#method.is_null)

So this line should be `if ptr.offset(i).is_null() {`

Reply:
Well, it is actually not a null pointer else the dereferencing would not work :)